artifact/checksum.go

```diff
@@ -98,6 +94,15 @@ func (c *Checksum) Checksum() []byte {
 	return checksum
 }
 
+func (c *Checksum) Verify() error {
+	sum := c.Checksum()
```

Comment:
Does this calculate the checksum for the whole file or it just grabs the running sum which has currently accumulated for the stream?

Reply:
The later. That's why this is important to call this function when we see EOF for any given part of the stream.